pkg/operator/ceph/pool/controller.go

```diff
@@ -497,3 +502,43 @@ func (r *ReconcileCephBlockPool) cancelMirrorMonitoring(cephBlockPool *cephv1.Ce
 		delete(r.blockPoolContexts, channelKey)
 	}
 }
+
+func (r *ReconcileCephBlockPool) disableMirroring(pool *cephv1.CephBlockPool) error {
+	mirrorInfo, err := cephclient.GetPoolMirroringInfo(r.context, r.clusterInfo, pool.Name)
+	if err != nil {
+		return errors.Wrapf(err, "failed to get mirroring info for the pool %q", pool.Name)
+	}
+
+	if mirrorInfo.Mode == "image" {
+		mirroredPools, err := cephclient.GetMirroredPoolImages(r.context, r.clusterInfo, pool.Name)
+		if err != nil {
+			return errors.Wrapf(err, "failed to list mirrored images for pool %q", pool.Name)
+		}
+
+		if len(*mirroredPools.Images) > 0 {
+			msg := fmt.Sprintf("there are images in the pool %q. Please manually disable mirroring for each image", pool.Name)
+			logger.Errorf(msg)
+			return errors.New(msg)
```

Comment:
This will cause the reconcile to fail again and again, until the user removes them manually. Seems like we should just log the error and not fail the reconcile. Let's discuss.